viewer/src/public/migration/Cognite3DViewer.ts

```diff
@@ -1042,6 +1045,82 @@ export class Cognite3DViewer {
     throw new NotSupportedInMigrationWrapperError('Cache is not supported');
   }
 
+  /**
+   * Creates a tool for attaching HTML elements to a 3D position and updates it's
+   * position/visibility as user moves the camera. This is useful to create HTML
+   * overlays to highlight information about key positions in the 3D model.
+   *
+   * Attached elements *must* have CSS style 'position: absolute'. It's also recommended
+   * in most cases to have styles 'pointerEvents: none' and 'touchAction: none' to avoid
+   * interfering with 3D navigation. Consider also applying 'transform: translate(-50%, -50%)'
+   * to anchor the center of the element rather than the top-left corner. In some cases the
+   * `zIndex`-attribute is necessary for the element to appear on top of the viewer.
+   *
+   * @example
+   * ```js
+   * const el = document.createElement('div');
+   * el.style.position = 'absolute'; // Required!
+   * // Anchor to center of element
+   * el.style.transform = 'translate(-50%, -50%)';
+   * // Avoid being target for events
+   * el.style.pointerEvents = 'none;
+   * el.style.touchAction = 'none';
+   *
+   * el.style.color = 'red';
+   * el.innerHtml = '<h1>Overlay</h1>';
+   *
+   * const overlayTool = viewer.createHtmlOverlays();
+   * overlayTool.attachHtmlElement(el, new THREE.Vector3(10, 10, 10));
+   * // ...
+   * overlayTool.removeHtmlElement(el);
+   * // or, to remove all attached elements:
+   * overlayTool.dispose();
+   * ```
+   */
+  createHtmlOverlayTool(): HtmlOverlayTool {
+    const overlays = new HtmlOverlayTool(this.domElement, this.renderer, this.camera);
+    this.registerTool(overlays);
+    return overlays;
+  }
+
+  /**
+   * Registers a new tool. This makes the tool get updates from the viewer.
+   * The tool will be unregistered when it's diposed.
+   * @param tool Tool to register.
+   */
+  private registerTool(tool: Cognite3DViewerTool): void {
+    if (this._tools.has(tool)) {
+      throw new Error(`Tool ${tool} is already registered`);
+    }
+    const onDisposed: DisposedDelegate = () => this.unregisterTool(tool);
```

Comment:
why do you need that DisposedDelegate type at all?

Reply:
Maybe its a bit overkill, but I like the idea that you can change the signature one place and have it be updated automatically or result in at least better error messages for references that doesn't work.

Reply:
> @larsmoa at least requesting changes for suggestions to be committed. Config changes seem irrelevant.
> 
> How did you fix your circular dependency issue btw? What was wrong? I don't think you made it work by changing those configs.

No, the configs was things I tested and forgot to remove - sorry. I fixed the error using `madge` to create a graph of dependencies and found some circular stuff. Never really understood exactly what the issue was, but fixed by changing imports to ensure the circual deps was removed.

> 
> About the new API. I think it's an interesting choice to manage additional tools, to avoid Viewer cluttering. Good job. Though in my opinion, I think biderectional connection for disposal might be good thing. Not only tools should notify the viewer that they are disposed, but the viewer should destroy those tools which are not relevant anymore after the viewer is disposed.